### PR TITLE
Added some additional JavaScript Style Guidelines

### PR DESCRIPTION
**Pull request by phillipvose**
Nabbed some guidelines from Air'b'n'b's website: https://github.com/airbnb/javascript

### CODE REVIEW

File: javascript.md [hunk 1]
```diff
@@ -2,13 +2,36 @@
...
+
+The best way to go about writing JavaScript for our site is to make it _readable_ and use the `KISS methodology`.
+
 * Use spaces instead of tabs with a 2 space indent
-* Use multiple lines for multiple selectors
-* Use double quotes `"` instead of single quotes `'` around strings where possible
-* Our naming convention is camelCase (https://en.wikipedia.org/wiki/CamelCase) when naming variables and methods
-* Files should be named in Snake_Case to match the Ruby File Structure
+* Our naming convention for variables and methods is [camelCase](https://en.wikipedia.org/wiki/CamelCase)
+* Files should be named in [Snake_Case](https://en.wikipedia.org/wiki/Snake_case) to match the Ruby File Structure
```
Comment (jadobson) [inline javascript.md]: Can you lowercase the `Snake_Case` here, it's confusing, as `snake_case` is supposed to be lowercase.

File: javascript.md [hunk 2]
```diff
@@ -2,13 +2,36 @@
...
+
+### Browser Support
+
+We support as far back as Internet Explorer 8 so please make sure any later ECMAScript specific functions you use will work with this browser without throwing an error.
+
+### Strings
+
+* We prefer the use of double quotes `"` over single quotes `'` around strings unless there is good reason
+* Strings should be kept as brief as possible and should line wrap after 80 characters
+* Where possible use translations provided through `i18n` instead of hard-coding text as a string to allow globalisation
```
Comment (jadobson) [inline javascript.md]: This should be `I18n`

File: javascript.md [hunk 3]
```diff
@@ -2,13 +2,36 @@
...
+* Never use `eval()` on a string, it opens too many vulnerabilities
+
+### jQuery
+
+* Prefix jQuery object variables with a `$`
+* Use chaining where possible, spanned over several lines to improve readability
+* __Don't__ use jQuery animations or show/hide functions __ever__ as the jQuery animation library is pretty slow, use CSS to achieve the same effect instead
+* Cache your jQuery lookups, it makes things quicker
```
Comment (jadobson) [inline javascript.md]: Could we expand on this? Probably something like:

> Cache your jQuery lookups if you're using them more than once, it makes things quicker

I wonder if it's worth adding something about chaining here too? 

> Make use of jQuery method chaining where possible

**Comment by jadobson** [comment]
I wonder if it's worth adding some examples to some of these points? For example, method chaining,

```
this.$(".element")
  .method1()
  .method2()
  .method3();
```

**Comment by jstarmx** [comment]
:+1:

File: javascript.md [hunk 4]
```diff
@@ -2,13 +2,42 @@
...
+### Strings
+
+* We prefer the use of double quotes `"` over single quotes `'` around strings unless there is good reason
+* Strings should be kept as brief as possible and should line wrap after 80 characters
+* Where possible use translations provided through `I18n` instead of hard-coding text as a string to allow globalisation
+* Never use `eval()` on a string, it opens too many vulnerabilities
+
+### jQuery
+
+* Prefix jQuery object variables with a `$`
```
Comment (jadobson) [inline javascript.md]: Could you put an example at the end of this, i.e.

> Prefix jQuery object variables with a `$`, e.g. `$element`

File: javascript.md [hunk 5]
```diff
@@ -2,13 +2,42 @@
...
+* Use chaining where possible, spanned over several lines to improve readability
+```
+this.$(".element")
+  .method1()
+  .method2()
+  .method3();
+```
+* __Don't__ use jQuery animations or show/hide functions __ever__ as the jQuery animation library is pretty slow, use CSS to achieve the same effect instead
+* Cache your jQuery lookups if you're using them more than once, it makes things quicker
+* Avoid using jQuery plugins where possible
```
Comment (jadobson) [inline javascript.md]: I wonder if it's worth expanding this line a little bit, just to say why. Something like

> Avoid using jQuery plugins where possible. Most of the time the functionality we need is a fraction of what plugins provide. We don't need unnecessary code bloat.

**Comment by jadobson** [comment]
:+1: 

A lot more needed on this, but it's a good start!

**Pull request merged**